GH-133336: Remove reserved ``-J`` flag for Jython (#133444)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2025-05-05-15-33-35.gh-issue-133336.miffFi.rst

> :option:`!-J` is no longer reserved for use by Jython.

Reviewer: ```suggestion
Remove the :option:`!-J` option reserved for use by Jython.
```
Maybe a bit clearer?

Author: I prefer the current wording, we aren't removing the flag, as it did nothing, but instead removing the reservation of the flag.
